cpp/tests/utilities/cudf_test_fixtures.h

```diff
@@ -48,3 +50,49 @@ struct GdfTest : public ::testing::Test
         ASSERT_RMM_SUCCEEDED( rmmFinalize() );
     }
 };
+
+/**
+* @brief Environment for google tests that creates/deletes temporary directory 
+* for each test program and provides path of filenames
+* 
+* TempDirTestEnvironment* const temp_env = static_cast<TempDirTestEnvironment*>(
+*   ::testing::AddGlobalTestEnvironment(new TempDirTestEnvironment));
+*/
+struct TempDirTestEnvironment : public ::testing::Environment
+{
+    char* tmpdir;
+
+    void SetUp() {
+        tmpdir=new char[30];
+        strcpy(tmpdir, "/tmp/gtest.XXXXXX");
+        tmpdir=mkdtemp(tmpdir);
+        strcat(tmpdir, "/");
+    }
+
+    void TearDown() {
+        //TODO: should use std::filesystem instead, once C++17 support added
+        nftw(tmpdir, rm_files, 10, FTW_DEPTH|FTW_MOUNT|FTW_PHYS);
+        delete[] tmpdir;
+        tmpdir=NULL;
+    }
+
+    static int rm_files(const char *pathname, const struct stat *sbuf, int type, struct FTW *ftwb)
+    {
+        return remove(pathname);
+    }
+
+    /**
+    * @brief get temporary path of filename for this test program
+    *
+    * @param fname filename or relative path
+    *
+    * @return absolute temporary file path of filename
+    */
+    char* get_temp_filename(const char *fname)
+    {
+        char* tmpfname = new char[strlen(tmpdir)+strlen(fname)+1];
```

Comment:
Where is the memory for the tmpfname deallocated?

Reply:
Good point. This should all be std::string() anyway, IMO. Just store tmpdir as a std::string in the struct, and have a function to return that. The caller can just concatenate it with their own string using C++ strings...  That way any temporary memory allocation is local to the use.

Reply:
For some reason, the tests used char*. I assumed it was to keep tests C-compatible. I will use std::string going forward.

Reply:
Yeah, probably because the interface takes in `char*` strings and there wasn't any need for string manipulation before. Switching to `std::string` is fine.